### ISSUE TITLE
Bundling fails when adding content to a bundle then attepting to push or download the bundle

### ISSUE THREAD

**Issue by craigWagner99** [dotCMS/core#19636]
v 5.3.9
As reported in https://dotcms.zendesk.com/agent/tickets/102965

Bundling fails when you add content to a bundle and try to push or download the bundle

**To Reproduce**
1. Create a new bundle of at least one item.
2. Use any publishing filter other than 'Only Selected Items' (Only Selected Items filter _does_ work)
3. Attempt to push publish or download bundle. 

Note: The content type this happens most often with is self-related.  The other content type it happens on is related to a self-related content type.  So there's a chance this may have something to do with the self-related content types.

Stacktrace of error:

```
com.dotcms.publishing.DotBundleException: com.dotcms.publisher.util.DependencyManager : generate()Problem occured while publishing file: Unable to pull content
	at com.dotcms.enterprise.publishing.remote.bundler.DependencyBundler.generate(SourceFile:71) ~[ee-5.3.9.jar:?]
	at com.dotcms.publishing.PublisherAPIImpl.publish_aroundBody0(PublisherAPIImpl.java:124) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.publishing.PublisherAPIImpl$AjcClosure1.run(PublisherAPIImpl.java:1) ~[dotcms_5.3.9_999999.jar:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
	at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.aspects.interceptors.CloseDBIfOpenedMethodInterceptor.invoke(CloseDBIfOpenedMethodInterceptor.java:29) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.aspects.aspectj.CloseDBIfOpenedAspect.invoke(CloseDBIfOpenedAspect.java:41) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.publishing.PublisherAPIImpl.publish(PublisherAPIImpl.java:44) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.publishing.PublisherAPIImpl.publish(PublisherAPIImpl.java:39) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.publisher.business.PublisherQueueJob.execute_aroundBody0(PublisherQueueJob.java:185) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.publisher.business.PublisherQueueJob$AjcClosure1.run(PublisherQueueJob.java:1) ~[dotcms_5.3.9_999999.jar:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
	at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.aspects.interceptors.CloseDBIfOpenedMethodInterceptor.invoke(CloseDBIfOpenedMethodInterceptor.java:29) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.aspects.aspectj.CloseDBIfOpenedAspect.invoke(CloseDBIfOpenedAspect.java:41) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.publisher.business.PublisherQueueJob.execute(PublisherQueueJob.java:112) ~[dotcms_5.3.9_999999.jar:?]
	at org.quartz.core.JobRunShell.run(JobRunShell.java:223) ~[dot.quartz-all-1.8.6_2.jar:?]
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:549) ~[dot.quartz-all-1.8.6_2.jar:?]
Caused by: com.dotcms.publishing.DotBundleException: com.dotcms.publisher.util.DependencyManager : generate()Problem occured while publishing file: Unable to pull content
	at com.dotcms.publisher.util.DependencyManager.setContentDependencies(DependencyManager.java:1351) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.publisher.util.DependencyManager.setDependencies(DependencyManager.java:349) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.enterprise.publishing.remote.bundler.DependencyBundler.generate(SourceFile:68) ~[ee-5.3.9.jar:?]
	... 17 more
Caused by: com.dotmarketing.business.DotStateException: Problem occured while publishing file
	at com.dotcms.publisher.util.DependencyManager.processList(DependencyManager.java:1226) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.publisher.util.DependencyManager.setContentDependencies(DependencyManager.java:1348) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.publisher.util.DependencyManager.setDependencies(DependencyManager.java:349) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.enterprise.publishing.remote.bundler.DependencyBundler.generate(SourceFile:68) ~[ee-5.3.9.jar:?]
	... 17 more
20:36:00.913  ERROR business.PublisherQueueJob - Unable to publish Bundle '01EQVD2R9BJ10035YBK9G4YK6B': com.dotcms.publisher.util.DependencyManager : generate()Problem occured while publishing file: Unable to pull content
com.dotcms.publishing.DotPublishingException: com.dotcms.publisher.util.DependencyManager : generate()Problem occured while publishing file: Unable to pull content
	at com.dotcms.publishing.PublisherAPIImpl.publish_aroundBody0(PublisherAPIImpl.java:152) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.publishing.PublisherAPIImpl$AjcClosure1.run(PublisherAPIImpl.java:1) ~[dotcms_5.3.9_999999.jar:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
	at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.aspects.interceptors.CloseDBIfOpenedMethodInterceptor.invoke(CloseDBIfOpenedMethodInterceptor.java:29) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.aspects.aspectj.CloseDBIfOpenedAspect.invoke(CloseDBIfOpenedAspect.java:41) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.publishing.PublisherAPIImpl.publish(PublisherAPIImpl.java:44) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.publishing.PublisherAPIImpl.publish(PublisherAPIImpl.java:39) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.publisher.business.PublisherQueueJob.execute_aroundBody0(PublisherQueueJob.java:185) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.publisher.business.PublisherQueueJob$AjcClosure1.run(PublisherQueueJob.java:1) ~[dotcms_5.3.9_999999.jar:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
	at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.aspects.interceptors.CloseDBIfOpenedMethodInterceptor.invoke(CloseDBIfOpenedMethodInterceptor.java:29) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.aspects.aspectj.CloseDBIfOpenedAspect.invoke(CloseDBIfOpenedAspect.java:41) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.publisher.business.PublisherQueueJob.execute(PublisherQueueJob.java:112) ~[dotcms_5.3.9_999999.jar:?]
	at org.quartz.core.JobRunShell.run(JobRunShell.java:223) ~[dot.quartz-all-1.8.6_2.jar:?]
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:549) ~[dot.quartz-all-1.8.6_2.jar:?]
Caused by: com.dotcms.publishing.DotBundleException: com.dotcms.publisher.util.DependencyManager : generate()Problem occured while publishing file: Unable to pull content
	at com.dotcms.enterprise.publishing.remote.bundler.DependencyBundler.generate(SourceFile:71) ~[ee-5.3.9.jar:?]
	at com.dotcms.publishing.PublisherAPIImpl.publish_aroundBody0(PublisherAPIImpl.java:124) ~[dotcms_5.3.9_999999.jar:?]
	... 16 more
Caused by: com.dotcms.publishing.DotBundleException: com.dotcms.publisher.util.DependencyManager : generate()Problem occured while publishing file: Unable to pull content
	at com.dotcms.publisher.util.DependencyManager.setContentDependencies(DependencyManager.java:1351) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.publisher.util.DependencyManager.setDependencies(DependencyManager.java:349) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.enterprise.publishing.remote.bundler.DependencyBundler.generate(SourceFile:68) ~[ee-5.3.9.jar:?]
	at com.dotcms.publishing.PublisherAPIImpl.publish_aroundBody0(PublisherAPIImpl.java:124) ~[dotcms_5.3.9_999999.jar:?]
	... 16 more
Caused by: com.dotmarketing.business.DotStateException: Problem occured while publishing file
	at com.dotcms.publisher.util.DependencyManager.processList(DependencyManager.java:1226) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.publisher.util.DependencyManager.setContentDependencies(DependencyManager.java:1348) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.publisher.util.DependencyManager.setDependencies(DependencyManager.java:349) ~[dotcms_5.3.9_999999.jar:?]
	at com.dotcms.enterprise.publishing.remote.bundler.DependencyBundler.generate(SourceFile:68) ~[ee-5.3.9.jar:?]
	at com.dotcms.publishing.PublisherAPIImpl.publish_aroundBody0(PublisherAPIImpl.java:124) ~[dotcms_5.3.9_999999.jar:?]
	... 16 more
```

**Comment by john-thomas-dotcms** [comment]
The customer provided a DB dump, so we can attempt to reproduce the issue on a local system.  The dump is in the support ticket; if you don't have access to the ticket, please contact either Craig or me for a copy.

**Comment by nollymar** [comment]
**Note to QA:** test with a custom database

**Comment by nollymar** [comment]
Needs work: An exception is still thrown when I try to download a bundle: 

```
[class com.dotcms.publishing.GenerateBundlePublisher]
12:45:57.246  INFO  util.PushPublishLogger - class com.dotcms.publishing.PublisherAPIImpl : [BundleID: 01EVCCV1EK5SVFSB2C1EP75WPQ] Started Publishing Task
12:45:57.247  INFO  publishing.PublisherAPIImpl - Writing bundle.xml file
12:45:57.367  INFO  publishing.PublisherAPIImpl - Start of Bundler: DependencyBundler
12:45:57.371  INFO  util.DependencyManager - PublisherFilter { excludeClassesSet = [] , excludeDependencyClassesSet = [] , excludeQueryIds = [] , excludeDependencyQueryIds = [] , relationships = true , dependencies = true}
12:45:57.936  ERROR publishing.PublisherAPIImpl - com.dotcms.publisher.util.DependencyManager : generate()Problem occured while publishing file:Invalid UUID string: /images/markets/minneapolis-mn-main.jpg: Unable to pull content
com.dotcms.publishing.DotBundleException: com.dotcms.publisher.util.DependencyManager : generate()Problem occured while publishing file:Invalid UUID string: /images/markets/minneapolis-mn-main.jpg: Unable to pull content
	at com.dotcms.enterprise.publishing.remote.bundler.DependencyBundler.generate(SourceFile:71) ~[ee-21.01.jar:?]
	at com.dotcms.publishing.PublisherAPIImpl.publish_aroundBody0(PublisherAPIImpl.java:123) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publishing.PublisherAPIImpl$AjcClosure1.run(PublisherAPIImpl.java:1) ~[dotcms_21.01_1e5299d.jar:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
	at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.aspects.interceptors.CloseDBIfOpenedMethodInterceptor.invoke(CloseDBIfOpenedMethodInterceptor.java:29) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.aspects.aspectj.CloseDBIfOpenedAspect.invoke(CloseDBIfOpenedAspect.java:41) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publishing.PublisherAPIImpl.publish(PublisherAPIImpl.java:43) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publishing.PublisherAPIImpl.publish(PublisherAPIImpl.java:38) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publisher.bundle.business.BundleAPIImpl.generateBundleDirectory_aroundBody30(BundleAPIImpl.java:410) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publisher.bundle.business.BundleAPIImpl$AjcClosure31.run(BundleAPIImpl.java:1) ~[dotcms_21.01_1e5299d.jar:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
	at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.aspects.interceptors.CloseDBIfOpenedMethodInterceptor.invoke(CloseDBIfOpenedMethodInterceptor.java:29) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.aspects.aspectj.CloseDBIfOpenedAspect.invoke(CloseDBIfOpenedAspect.java:41) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publisher.bundle.business.BundleAPIImpl.generateBundleDirectory(BundleAPIImpl.java:405) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publisher.bundle.business.BundleAPIImpl.generateTarGzipBundleFile_aroundBody32(BundleAPIImpl.java:423) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publisher.bundle.business.BundleAPIImpl$AjcClosure33.run(BundleAPIImpl.java:1) ~[dotcms_21.01_1e5299d.jar:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
	at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.aspects.interceptors.CloseDBIfOpenedMethodInterceptor.invoke(CloseDBIfOpenedMethodInterceptor.java:29) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.aspects.aspectj.CloseDBIfOpenedAspect.invoke(CloseDBIfOpenedAspect.java:41) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publisher.bundle.business.BundleAPIImpl.generateTarGzipBundleFile(BundleAPIImpl.java:422) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.rest.BundleResource$BundleGenerator.run(BundleResource.java:753) ~[dotcms_21.01_1e5299d.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_211]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_211]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_211]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_211]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_211]
Caused by: com.dotcms.publishing.DotBundleException: com.dotcms.publisher.util.DependencyManager : generate()Problem occured while publishing file:Invalid UUID string: /images/markets/minneapolis-mn-main.jpg: Unable to pull content
	at com.dotcms.publisher.util.DependencyManager.setContentDependencies(DependencyManager.java:1371) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publisher.util.DependencyManager.setDependencies(DependencyManager.java:345) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.enterprise.publishing.remote.bundler.DependencyBundler.generate(SourceFile:68) ~[ee-21.01.jar:?]
	... 28 more
Caused by: com.dotmarketing.business.DotStateException: Problem occured while publishing file:Invalid UUID string: /images/markets/minneapolis-mn-main.jpg
	at com.dotcms.publisher.util.DependencyManager.processList(DependencyManager.java:1240) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publisher.util.DependencyManager.setContentDependencies(DependencyManager.java:1368) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publisher.util.DependencyManager.setDependencies(DependencyManager.java:345) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.enterprise.publishing.remote.bundler.DependencyBundler.generate(SourceFile:68) ~[ee-21.01.jar:?]
	... 28 more
Caused by: java.lang.IllegalArgumentException: Invalid UUID string: /images/markets/minneapolis-mn-main.jpg
	at java.util.UUID.fromString(UUID.java:194) ~[?:1.8.0_211]
	at com.dotmarketing.beans.Identifier.setId(Identifier.java:97) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotmarketing.business.IdentifierFactoryImpl.build404(IdentifierFactoryImpl.java:126) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotmarketing.business.IdentifierFactoryImpl.find(IdentifierFactoryImpl.java:476) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotmarketing.business.IdentifierAPIImpl.find_aroundBody4(IdentifierAPIImpl.java:83) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotmarketing.business.IdentifierAPIImpl$AjcClosure5.run(IdentifierAPIImpl.java:1) ~[dotcms_21.01_1e5299d.jar:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
	at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.aspects.interceptors.CloseDBIfOpenedMethodInterceptor.invoke(CloseDBIfOpenedMethodInterceptor.java:29) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.aspects.aspectj.CloseDBIfOpenedAspect.invoke(CloseDBIfOpenedAspect.java:41) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotmarketing.business.IdentifierAPIImpl.find(IdentifierAPIImpl.java:82) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publisher.util.DependencyManager.processList(DependencyManager.java:1234) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publisher.util.DependencyManager.setContentDependencies(DependencyManager.java:1368) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publisher.util.DependencyManager.setDependencies(DependencyManager.java:345) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.enterprise.publishing.remote.bundler.DependencyBundler.generate(SourceFile:68) ~[ee-21.01.jar:?]
	... 28 more
12:45:57.936  ERROR business.BundleAPIImpl - com.dotcms.publisher.util.DependencyManager : generate()Problem occured while publishing file:Invalid UUID string: /images/markets/minneapolis-mn-main.jpg: Unable to pull content
com.dotcms.publishing.DotPublishingException: com.dotcms.publisher.util.DependencyManager : generate()Problem occured while publishing file:Invalid UUID string: /images/markets/minneapolis-mn-main.jpg: Unable to pull content
	at com.dotcms.publishing.PublisherAPIImpl.publish_aroundBody0(PublisherAPIImpl.java:151) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publishing.PublisherAPIImpl$AjcClosure1.run(PublisherAPIImpl.java:1) ~[dotcms_21.01_1e5299d.jar:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
	at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.aspects.interceptors.CloseDBIfOpenedMethodInterceptor.invoke(CloseDBIfOpenedMethodInterceptor.java:29) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.aspects.aspectj.CloseDBIfOpenedAspect.invoke(CloseDBIfOpenedAspect.java:41) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publishing.PublisherAPIImpl.publish(PublisherAPIImpl.java:43) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publishing.PublisherAPIImpl.publish(PublisherAPIImpl.java:38) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publisher.bundle.business.BundleAPIImpl.generateBundleDirectory_aroundBody30(BundleAPIImpl.java:410) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publisher.bundle.business.BundleAPIImpl$AjcClosure31.run(BundleAPIImpl.java:1) ~[dotcms_21.01_1e5299d.jar:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
	at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.aspects.interceptors.CloseDBIfOpenedMethodInterceptor.invoke(CloseDBIfOpenedMethodInterceptor.java:29) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.aspects.aspectj.CloseDBIfOpenedAspect.invoke(CloseDBIfOpenedAspect.java:41) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publisher.bundle.business.BundleAPIImpl.generateBundleDirectory(BundleAPIImpl.java:405) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publisher.bundle.business.BundleAPIImpl.generateTarGzipBundleFile_aroundBody32(BundleAPIImpl.java:423) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publisher.bundle.business.BundleAPIImpl$AjcClosure33.run(BundleAPIImpl.java:1) ~[dotcms_21.01_1e5299d.jar:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
	at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.aspects.interceptors.CloseDBIfOpenedMethodInterceptor.invoke(CloseDBIfOpenedMethodInterceptor.java:29) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.aspects.aspectj.CloseDBIfOpenedAspect.invoke(CloseDBIfOpenedAspect.java:41) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publisher.bundle.business.BundleAPIImpl.generateTarGzipBundleFile(BundleAPIImpl.java:422) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.rest.BundleResource$BundleGenerator.run(BundleResource.java:753) ~[dotcms_21.01_1e5299d.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_211]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_211]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_211]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_211]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_211]
Caused by: com.dotcms.publishing.DotBundleException: com.dotcms.publisher.util.DependencyManager : generate()Problem occured while publishing file:Invalid UUID string: /images/markets/minneapolis-mn-main.jpg: Unable to pull content
	at com.dotcms.enterprise.publishing.remote.bundler.DependencyBundler.generate(SourceFile:71) ~[ee-21.01.jar:?]
	at com.dotcms.publishing.PublisherAPIImpl.publish_aroundBody0(PublisherAPIImpl.java:123) ~[dotcms_21.01_1e5299d.jar:?]
	... 27 more
Caused by: com.dotcms.publishing.DotBundleException: com.dotcms.publisher.util.DependencyManager : generate()Problem occured while publishing file:Invalid UUID string: /images/markets/minneapolis-mn-main.jpg: Unable to pull content
	at com.dotcms.publisher.util.DependencyManager.setContentDependencies(DependencyManager.java:1371) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publisher.util.DependencyManager.setDependencies(DependencyManager.java:345) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.enterprise.publishing.remote.bundler.DependencyBundler.generate(SourceFile:68) ~[ee-21.01.jar:?]
	at com.dotcms.publishing.PublisherAPIImpl.publish_aroundBody0(PublisherAPIImpl.java:123) ~[dotcms_21.01_1e5299d.jar:?]
	... 27 more
Caused by: com.dotmarketing.business.DotStateException: Problem occured while publishing file:Invalid UUID string: /images/markets/minneapolis-mn-main.jpg
	at com.dotcms.publisher.util.DependencyManager.processList(DependencyManager.java:1240) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publisher.util.DependencyManager.setContentDependencies(DependencyManager.java:1368) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publisher.util.DependencyManager.setDependencies(DependencyManager.java:345) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.enterprise.publishing.remote.bundler.DependencyBundler.generate(SourceFile:68) ~[ee-21.01.jar:?]
	at com.dotcms.publishing.PublisherAPIImpl.publish_aroundBody0(PublisherAPIImpl.java:123) ~[dotcms_21.01_1e5299d.jar:?]
	... 27 more
Caused by: java.lang.IllegalArgumentException: Invalid UUID string: /images/markets/minneapolis-mn-main.jpg
	at java.util.UUID.fromString(UUID.java:194) ~[?:1.8.0_211]
	at com.dotmarketing.beans.Identifier.setId(Identifier.java:97) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotmarketing.business.IdentifierFactoryImpl.build404(IdentifierFactoryImpl.java:126) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotmarketing.business.IdentifierFactoryImpl.find(IdentifierFactoryImpl.java:476) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotmarketing.business.IdentifierAPIImpl.find_aroundBody4(IdentifierAPIImpl.java:83) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotmarketing.business.IdentifierAPIImpl$AjcClosure5.run(IdentifierAPIImpl.java:1) ~[dotcms_21.01_1e5299d.jar:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
	at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.aspects.interceptors.CloseDBIfOpenedMethodInterceptor.invoke(CloseDBIfOpenedMethodInterceptor.java:29) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.aspects.aspectj.CloseDBIfOpenedAspect.invoke(CloseDBIfOpenedAspect.java:41) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotmarketing.business.IdentifierAPIImpl.find(IdentifierAPIImpl.java:82) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publisher.util.DependencyManager.processList(DependencyManager.java:1234) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publisher.util.DependencyManager.setContentDependencies(DependencyManager.java:1368) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.publisher.util.DependencyManager.setDependencies(DependencyManager.java:345) ~[dotcms_21.01_1e5299d.jar:?]
	at com.dotcms.enterprise.publishing.remote.bundler.DependencyBundler.generate(SourceFile:68) ~[ee-21.01.jar:?]
	at com.dotcms.publishing.PublisherAPIImpl.publish_aroundBody0(PublisherAPIImpl.java:123) ~[dotcms_21.01_1e5299d.jar:?]
	... 27 more
```

**Comment by erickgonzalez** [comment]
The error @nollymar was getting was because of bad data (value of an image field was a path instead of a UUID, at first we thought about a possible fix, but then realized that is not gonna work in this case, because the image of the path no longer exists).
The fix for the bad data was added to the zendesk ticket.

**Comment by jcastro-dotcms** [comment]
The official code changes have been included in the dotCMS 5.3.8.4 LTS release only. The 5.2.8.4 LTS release is several commits behind, which made the code back-port not possible.
